test/lit_tests/concatreshape2.mlir

```diff
@@ -0,0 +1,23 @@
+// RUN: enzymexlamlir-opt --pass-pipeline="builtin.module(enzyme-hlo-generate-td{patterns=concat_reshape},transform-interpreter,enzyme-hlo-remove-transform)" %s | FileCheck %s
+// RUN: enzymexlamlir-opt --pass-pipeline="builtin.module(enzyme-hlo-generate-td{patterns=concat_reshape},transform-interpreter,enzyme-hlo-remove-transform)" %s |  stablehlo-translate - --interpret
+
+module {
+    func.func @main() -> tensor<2x4x2xf32> {
+        %0 = stablehlo.constant dense<[[1.0, 2.0, 3.0, 4.0], [5.0, 6.0, 7.0, 8.0]]> : tensor<2x4xf32>
+        %1 = stablehlo.reshape %0: (tensor<2x4xf32>) -> tensor<4x2xf32>
```

Comment:
this I'm less sure of -- it relies on the layout of the operand being reshaped to be correct?

Reply:
are we not guaranteed a row-major layout here?

Reply:
no we are not, there's a layout assignment done at xla level

Reply:
Ah fair. Then this would be an incorrect pattern if the layout isn't guaranteed. Though we might want to think of some way to do this one, since it is essentially concat along batch dimension